File: frontend/src/types/comment.ts
export interface ReplyToSummary {
  commentId: number
  userId: number
  username: string
  displayName: string
  deleted: boolean
}

export interface Comment {
  id: number
  postId: number
  userId: number
  username: string
  displayName: string
  content: string | null
  createdAt: string
  updatedAt: string
  deleted: boolean
  replyTo: ReplyToSummary | null
}

File: backend/src/main/java/com/example/murmur/comment/Comment.java
package com.example.murmur.comment;

import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Comment {

    private Long id;
    private Long postId;
    private Long userId;
    private Long parentCommentId;
    private String content;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private boolean deleted;
    private String username;
    private String displayName;
    private Long replyToCommentId;
    private Long replyToUserId;
    private String replyToUsername;
    private String replyToDisplayName;
    private boolean replyToDeleted;
}
